<launch>
  <arg name="node_name" default="bilateral_filter" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

  <arg name="kernel_size" default="7" doc="Defines the size of the kernel to be used" />
  <arg name="sigma_color" default="35" doc="Filter sigma in the color space. A larger value of the parameter means that farther colors within the pixel neighborhood will be mixed together, resulting in larger areas of semi-equal color." />
  <arg name="sigma_space" default="5" doc="Filter sigma in the coordinate space. A larger value of the parameter means that farther pixels will influence each other as long as their colors are close enough." />

  <!-- bilateral_filter.cpp -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="bilateral_filter" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="kernel_size" value="$(arg kernel_size)" />
    <param name="sigma_color" value="$(arg sigma_color)" />
    <param name="sigma_space" value="$(arg sigma_space)" />
  </node>
</launch>
